Build failure: pkgsLLVM.xorg.libX11

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `pkgsLLVM.xorg.libX11`

### Build log

```
libX11-x86_64-unknown-linux-gnu-1.8.9> checking if x86_64-unknown-linux-gnu-clang supports -Werror... yes
libX11-x86_64-unknown-linux-gnu-1.8.9> checking if x86_64-unknown-linux-gnu-clang supports -Werror=attributes... yes
libX11-x86_64-unknown-linux-gnu-1.8.9> Package xorg-macros was not found in the pkg-config search path.
libX11-x86_64-unknown-linux-gnu-1.8.9> Perhaps you should add the directory containing `xorg-macros.pc'
libX11-x86_64-unknown-linux-gnu-1.8.9> to the PKG_CONFIG_PATH environment variable
libX11-x86_64-unknown-linux-gnu-1.8.9> No package 'xorg-macros' found
libX11-x86_64-unknown-linux-gnu-1.8.9> checking whether make supports nested variables... (cached) yes
libX11-x86_64-unknown-linux-gnu-1.8.9> checking whether to build functional specifications... yes
libX11-x86_64-unknown-linux-gnu-1.8.9> checking for xmlto... no
libX11-x86_64-unknown-linux-gnu-1.8.9> configure: WARNING: xmlto not found - documentation targets will be skipped
libX11-x86_64-unknown-linux-gnu-1.8.9> checking for xsltproc... no
libX11-x86_64-unknown-linux-gnu-1.8.9> configure: WARNING: xsltproc not found - cannot transform XML documents
libX11-x86_64-unknown-linux-gnu-1.8.9> checking for X.Org SGML entities >= 1.10... no
libX11-x86_64-unknown-linux-gnu-1.8.9> checking how to run the C preprocessor... x86_64-unknown-linux-gnu-clang -E
libX11-x86_64-unknown-linux-gnu-1.8.9> checking for x86_64-unknown-linux-gnu-cpp... /nix/store/id84g2s5lmzkkmrmqb7yhrki5hvyymg6-x86_64-unknown-linux-gnu-clang-wrapper-17.0.6/bin/x86_64-unknown-linux-gnu-cpp
libX11-x86_64-unknown-linux-gnu-1.8.9> checking if /nix/store/id84g2s5lmzkkmrmqb7yhrki5hvyymg6-x86_64-unknown-linux-gnu-clang-wrapper-17.0.6/bin/x86_64-unknown-linux-gnu-cpp requires -undef... configure: error: /nix/store/id84g2s5lmzkkmrmqb7yhrki5hvyymg6-x86_64-unknown-linux-gnu-clang-wrapper-17.0.6/bin/x86_64-unknown-linux-gnu-cpp defines unix with or without -undef.  I don't know what to do
```

### Additional context

Discovered via the Nixpkgs LLVM Workspace project.

https://github.com/RossComputerGuy/nixpkgs-llvm-ws/runs/26297896092

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

No maintainers, CC last person who touched @sdht0

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.158, NixOS, 24.05 (Uakari), 24.05.20240506.6ff120a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/run/current-system/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/build-support/cc-wrapper/add-clang-cc-cflags-before.sh
+++ b/pkgs/build-support/cc-wrapper/add-clang-cc-cflags-before.sh
@@ -6,6 +6,6 @@ for p in "${params[@]}"; do
     esac
 done
 
-if $needsTarget; then
+if $needsTarget && [[ $0 != *cpp ]]; then
     extraBefore+=(-target @defaultTarget@ @machineFlags@)
 fi

--- a/pkgs/servers/x11/xorg/overrides.nix
+++ b/pkgs/servers/x11/xorg/overrides.nix
@@ -147,7 +147,8 @@ self: super:
   libX11 = super.libX11.overrideAttrs (attrs: {
     outputs = [ "out" "dev" "man" ];
     configureFlags = attrs.configureFlags or []
-      ++ malloc0ReturnsNullCrossFlag;
+      ++ malloc0ReturnsNullCrossFlag
+      ++ lib.optional (stdenv.targetPlatform.useLLVM or false) "ac_cv_path_RAWCPP=cpp";
     depsBuildBuild = [
       buildPackages.stdenv.cc
     ] ++ lib.optionals stdenv.hostPlatform.isStatic [

